test/runner/e2e_delete_rel_test.cpp

```diff
@@ -19,104 +19,316 @@ class DeleteRelTest : public DBTest {
+        ASSERT_EQ(
+            TestHelper::convertResultToString(*conn->query(query), true /* checkOutputOrder */),
+            expectedResult);
+    }
 
-    auto actualResult = TestHelper::convertResultToString(
-        *conn->query("MATCH (:animal)-[e:knows]->(:person) return e.length"),
-        true /* checkOutputOrder */);
-    // After insertion and deletion, we are expected to see only one new rel: animal52->person0.
-    vector<string> expectedResult;
-    for (auto i = 0; i < NUM_RELS_FOR_ANIMAL_KNOWS_PERSON; i++) {
-        expectedResult.push_back(to_string(i));
+    void deleteMultipleRels(bool isCommit, TransactionTestType transactionTestType) {
+        conn->beginWriteTransaction();
+        // Delete all rels from animal[10-49]->person0.
+        ASSERT_TRUE(conn->query("MATCH (a:animal)-[e:knows]->(p:person) WHERE a.ID >= 10 AND "
+                                "a.ID <= 49 AND p.ID = 0 DELETE e;")
+                        ->isSuccess());
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        vector<string> expectedResult;
+        if (isCommit) {
+            expectedResult = vector<string>{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "50"};
+        } else {
+            for (auto i = 0u; i <= 50; i++) {
+                expectedResult.push_back(to_string(i));
+            }
+        }
+        ASSERT_EQ(TestHelper::convertResultToString(
+                      *conn->query("MATCH (a:animal)-[e:knows]->(p:person) RETURN e.length"),
+                      true /* checkOutputOrder */),
+            expectedResult);
+    }
+
+    void deleteAllInsertedRels(bool isCommit, TransactionTestType transactionTestType) {
+        conn->beginWriteTransaction();
+        // Insert rels for animal[60-69]->person0.
+        for (auto i = 60; i < 70; i++) {
+            ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("animal", "person", i, 0 /* dstID */))
+                            ->isSuccess());
+        }
+        // Then delete those inserted rels.
+        for (auto i = 60; i < 70; i++) {
+            ASSERT_TRUE(conn->query(getDeleteKnowsRelQuery("animal", "person", i, 0 /* dstID */))
+                            ->isSuccess());
+        }
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        vector<string> expectedResult = {
+            "40", "41", "42", "43", "44", "45", "46", "47", "48", "49", "50"};
+        ASSERT_EQ(
+            TestHelper::convertResultToString(
+                *conn->query(
+                    "MATCH (a:animal)-[e:knows]->(p:person) WHERE a.ID >= 40 RETURN e.length"),
+                true /* checkOutputOrder */),
+            expectedResult);
     }
-    expectedResult.push_back(to_string(52));
-    ASSERT_EQ(actualResult, expectedResult);
+
+    void deleteLargeNumRelsFromLargeList(bool isCommit, TransactionTestType transactionTestType) {
+        conn->beginWriteTransaction();
+        conn->query(
```

Comment:
Is this deleting all rels or almost all rels? Maybe remove this test and add two tests that test the edge cases of deleting all rels from a small and large list.

Reply:
This test is to test deleting large number of rels from a large list.
Addded two tests for deleting all rels from small/large list.